File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/bridge/UIManagerProvider.java
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.react.bridge;

/**
 * @deprecated since this would be deprecated later as part of Stable APIs with bridge removal and
 *     not encouraged usage.
 */
@Deprecated
public interface UIManagerProvider {

  UIManager createUIManager(ReactApplicationContext context);
}

File: packages/react-native/ReactAndroid/src/main/java/com/facebook/react/bridge/UIManagerProvider.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.react.bridge

import com.facebook.react.common.annotations.internal.LegacyArchitecture
import com.facebook.react.common.annotations.internal.LegacyArchitectureLogLevel

/**
 * [UIManagerProvider] is used to create UIManager objects during the initialization of React
 * Native.
 */
@LegacyArchitecture(logLevel = LegacyArchitectureLogLevel.ERROR)
@Deprecated(
    message = "This class is part of Legacy Architecture and will be removed in a future release",
    level = DeprecationLevel.WARNING,
)
public fun interface UIManagerProvider {

  /* Provides a [com.facebook.react.bridge.UIManager] for the context received as a parameter. */
  public fun createUIManager(context: ReactApplicationContext): UIManager?
}
